src/data/list/defs.lean

```diff
@@ -633,6 +633,16 @@ list.decidable_pairwise
      erase_dup [1, 0, 2, 2, 1] = [0, 2, 1] -/
 def erase_dup [decidable_eq α] : list α → list α := pw_filter (≠)
 
+/-- Removes all adjacent duplicates in `a :: l`. -/
+def chain_dedup [decidable_eq α] : α → list α → list α
+| a [] := []
+| a (h :: l) := if a = h then chain_dedup h l else h :: chain_dedup h l
+
+/-- Removes all adjacent duplicates in a list. -/
+def chain'_dedup [decidable_eq α] : list α → list α
+| (h :: l) := h :: chain_dedup h l
+| [] := []
```

Comment:
I've seen the name `norep` for this before (or `uniq`, but I find that one confusing). Do you need the first definition for anything else? If not, you can fold it in:

```lean
def norep {α : Type*} [decidable_eq α] : list α → list α
| [] := []
| [a] := [a]
| (a :: bs@(b :: cs)) := if a = b then norep (a :: cs) else a :: norep bs
```
(basically from [this](https://hackage.haskell.org/package/regex-tdfa-1.3.1.1/docs/src/Text.Regex.TDFA.Common.html#norep) or [this](https://hackage.haskell.org/package/hw-mquery-0.2.1.0/docs/src/HaskellWorks.Data.MQuery.html#uniq))

If you do need both definitions, I'd be tempted to call the first `norep'` and the second `norep`.

Reply:
I'm happy to have `norep`, but we should keep the priming consistent with `chain` and `chain'`; I think that those are the wrong way round but we can fix that in a follow-up PR